src/math/bspline.cpp

```diff
@@ -22,6 +22,8 @@ Eigen::VectorXd interpolateAt(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int
   const int ndofs = xs.rows(); // number of dofs. Each dof needs it's own interpolant.
 
   // transform all times to the relative interval [0; 1]
+  PRECICE_ASSERT(ts(0) != ts(ts.size() - 1));
+  // PRECICE_ASSERT((t >= ts(0)) && (t <= ts(ts.size() - 1)), "t lies outside of ts!", t, ts); // Only allowed to use BSpline for interpolation, not extrapolation.  // @todo fix this! Currently leads to failures in tests.
```

Comment:
I think we should try to also use this assertion to avoid unintended B-Spline extrapolation. Besides this point we are getting closer to the finish line here.

Reply:
We can also possibly add these assertions later via #1775 in favor of merging this PR earlier. The possible bugs I mention #1775 already exist on develop. Whether we fix them here or in #1775 does not really matter.